docs/architecture/go_components.md

```diff
@@ -0,0 +1,117 @@
+
+```
+	cloud.google.com/go v0.56.0
+	code.cloudfoundry.org/clock v1.0.0 // indirect
+	collectd.org v0.5.0
+	github.com/Azure/azure-event-hubs-go/v3 v3.2.0
+	go.starlark.net v0.0.0-20210406145628-7a1108eaa012
+	go.uber.org/multierr v1.6.0 // indirect
+	golang.org/x/net v0.0.0-20210226172049-e18ecbb05110
+	k8s.io/api v0.20.4
+	modernc.org/sqlite v1.10.8
+```
+
+Notice the variety in paths? Google allows "vanity" package names (like `go.uber.org/multierr` - which are automatically redirected to the source by Google's proxy service.
+
+## Constructing coordinates
+
+In order to request a definition of a go component from ClearlyDefined (which will also queue up a harvest if ClearlyDefined does not have a definition for the component), a user will need to construct coordinates in the API call.
+
+### For a Maven component
+
+For example, in order to request a maven package, the API call would look like this:
+
+```bash
+curl -X GET https://clearlydefined.io/definitions/maven/mavencentral/com.apollographql.apollo3/apollo-runtime-jvm/3.0.0-dev14" -H "accept: */*"
+```
+
+In this case, the coordinates are maven/mavencentral/com.apollographql.apollo3/apollo-runtime-jvm/3.0.0-dev14
+
+Let's break this down:
+* **maven** is the **type** 
+* **mavencentral** is the **provider**
+* **com.apollographql.apollo3** is the **namespace**
+* **apollo-runtime-jvm** is the **name**
+* **3.0.0-dev14** is the **revision**
+
+When we run this API call:
+
+```bash
+curl -X GET https://clearlydefined.io/definitions/maven/mavencentral/com.apollographql.apollo3/apollo-runtime-jvm/3.0.0-dev14" -H "accept: */*"
+```
+
+It harvests the component from [here](https://repo1.maven.org/maven2/com/apollographql/apollo3/apollo-runtime-jvm/3.0.0-dev14/).
+
+Notice something in the url - the namespace is "com/apollographql/apollo3" - there are a lot of slashes in there. 
+
+The way we translate this namespace into coordinates is to replace the slashes in the namespace with periods.
+
+So "com/apollographql/apollo3" becomes "com.apollographql.apollo3".
+
+### For a Go component
+
+A Go component's coordinates will also need to consist of a type, provider, namespace, name, and revision.
+
+Go's support of vanity package names makes this a little different than other components. While there are a few different approaches we could take, we should map the coordinates to what go developers know the package as.
+
+* **type** should be go
+* **provider** should be "proxy.golang.org" (we may need to add other providers in the future)
+* **namespace** should be anything that appears before the package name
+* **name** should be the name of the package
+* **revision** should be the component revision (note - go components seem to have multiple ways of specifying revisons - sometimes version numbers, sometimes commit hashes)
+
+Like with Maven components, we translate namespaces that have multiple slashes (like `golang.org/x`) into namepaces with periods (`golang.org.x`).
```

Comment:
It might be worth defining what happens in the highly-unlikely case of a name collision with this conversion:
```
github.com/user/repo/pkg -> ["github.com.user.repo", "pkg"]
github.com/user.repo/pkg -> ["github.com.user.repo", "pkg"]
```

Reply:
Oy....that's a good point.

Reply:
Going to open a larger issue for this, since this affects Maven and GitLab components (at least) as well.

Reply:
Done! #863